src/Rings/mpoly-graded.jl

```diff
@@ -1346,61 +1346,103 @@ function homogeneous_component(R::MPolyDecRing, g::IntegerUnion)
   return homogeneous_component(R, grading_group(R)([g]))
 end
 
-function vector_space(K::AbstractAlgebra.Field, e::Vector{T}; target = nothing) where {T <:MPolyRingElem}
+
+@doc raw"""
+  vector_space(K::Field, polys::Vector{T}; target = nothing) where {T <: MPolyRingElem}
+
+Return a `K`-vector space `V` and an epimorphism from `V` onto the `K`-vector
+space spanned by the polynomials in `polys`.
+
+Note that all polynomials must have the same parent `R`, and `K` must be equal
+to `base_ring(R)`. The optional keyword argument `target` can be used to
```

Comment:
Given this requirement on `K`, I wonder why we even have it as function argument? Perhaps we should at least make it optional?

Thoughts @thofma @fieker ?

Reply:
indeed, and that's why we have the `target` KW argument. But even in that scenario we don't need `K` -- after all we have this in the code: `@assert base_ring(R) == K` (where `R` is either `target` or else `parent(polys[1])`)

Reply:
Ah oops, I misunderstood, you are right.